### PR TITLE
[test gap] Add memory exhaustion test

### LINKED ISSUE #3619: Need new test: Memory Exhaustion
When Kernel runs out of memory and hit oom event, kernel will panic and reboot the DUT. (need a test to validate).

### PR DESCRIPTION
**Pull request by lizhijianrd**
### Description of PR

Validate kernel will panic and reboot the DUT when kernel runs out of memory and hit oom event.

Summary:
Fixes #3619 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request

- [x] 202012

#### What is the motivation for this PR?

Validate kernel will panic and reboot the DUT when kernel runs out of memory and hit oom event.

#### How did you do it?

Use `tail /dev/zero` to run out of memory completely and check the DUT reboot as expected.

#### How did you verify/test it?

Run tests on vSONiC and physical DUTs.

### Documentation

*  #3619

### CODE REVIEW

File: tests/platform_tests/test_memory_exhaustion.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.platform.processes_utils import wait_critical_processes
+from tests.common.reboot import SONIC_SSH_PORT, SONIC_SSH_REGEX
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+
+def test_memory_exhaustion(duthosts, enum_frontend_dut_hostname, localhost):
```
Comment (Blueve) [inline tests/platform_tests/test_memory_exhaustion.py:13]: @liuh-80 Is this test expected after your OOM protection merged?

**Review by Blueve — CHANGES_REQUESTED** [review]

File: tests/platform_tests/test_memory_exhaustion.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    logging.info('waiting for ssh to drop on {}'.format(hostname))
+    res = localhost.wait_for(host=dut_ip,
+                             port=SONIC_SSH_PORT,
+                             state='absent',
+                             search_regex=SONIC_SSH_REGEX,
+                             delay=10,
+                             timeout=120,
+                             module_ignore_errors=True)
+    if res.is_failed or ('msg' in res and 'Timeout' in res['msg']):
+        raise Exception('DUT {} did not shutdown'.format(hostname))
```
[the remark this answers is not in the document]
Comment (lizhijianrd) [reply tests/platform_tests/test_memory_exhaustion.py]: @Blueve Thank you! I updated my code in commit https://github.com/Azure/sonic-mgmt/pull/5260/commits/6c0a36a4c815b5e1f3d1d2364bb29ee69f54dadf. I hope this problem is fixed.

File: tests/platform_tests/test_memory_exhaustion.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    logging.info('waiting for ssh to startup on {}'.format(hostname))
+    res = localhost.wait_for(host=dut_ip,
+                             port=SONIC_SSH_PORT,
+                             state='started',
+                             search_regex=SONIC_SSH_REGEX,
+                             delay=10,
+                             timeout=120,
+                             module_ignore_errors=True)
+    if res.is_failed or ('msg' in res and 'Timeout' in res['msg']):
+        raise Exception('DUT {} did not startup'.format(hostname))
```
[the remark this answers is not in the document]
Comment (lizhijianrd) [reply tests/platform_tests/test_memory_exhaustion.py]: @Blueve Thank you! I updated my code in commit https://github.com/Azure/sonic-mgmt/pull/5260/commits/6c0a36a4c815b5e1f3d1d2364bb29ee69f54dadf. I hope this problem is fixed.

File: tests/platform_tests/test_memory_exhaustion.py (hunk 1 above)
Comment (Blueve) [reply tests/platform_tests/test_memory_exhaustion.py:13]: Let's sync this later. Before the feature merged, the test should be valid

Comment (Blueve) [reply tests/platform_tests/test_memory_exhaustion.py:13]: Confirmed with Hua, this is still a expected test case.

File: tests/platform_tests/test_memory_exhaustion.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import logging
```
[the remark this answers is not in the document]
Comment (Blueve) [reply tests/platform_tests/test_memory_exhaustion.py:1]: Let's move this test to other place in a new PR if need.

**Review by Blueve — APPROVED** [review]

**Comment by wangxin** [comment]
This PR should not go to 202012 branch. I have reverted it in 202012 branch.

**Pull request merged**